specification/app/resource-manager/Microsoft.App/preview/2024-02-02-preview/ContainerApps.json

```diff
@@ -886,6 +903,36 @@
         }
       }
     },
+    "LoglevelSetting": {
+      "description": "Log level settings for java workloads.",
+      "type": "object",
+      "required": [
+        "logger",
+        "level"
+      ],
+      "properties": {
+        "logger": {
+          "description": "Logger name.",
+          "type": "string"
+        },
+        "level": {
+          "description": "The specified logger's log level.",
+          "type": "string",
+          "enum": [
```

Comment:
Standard log levels vary from different log frameworks, e.g., [log4j2](https://logging.apache.org/log4j/2.x/javadoc/log4j-api/org/apache/logging/log4j/Level.html), [logback](https://www.slf4j.org/api/org/slf4j/event/Level.html), and some supports [level customization](https://logging.apache.org/log4j/2.x/manual/customloglevels.html). Perhaps we can define `level` just as a string instead of an `enum`?

Reply:
agree, good suggestion!

Reply:
2 points from my side:

1. If we use `string` type instead of `enum` type, the log level input experience on Azure portal must use text box (not sure this limitation is correct or not), but I expect the experience should be dropdown box because it is a better experience for customers.

2. I think we don't need to consider customization, just use standard log level like Spring Boot Admin. SBA use `OFF`, 'ERROR', 'INFO', 'DEBUG' and 'TRACE' to achieve log level update.

Reply:
Have similar preference here. One potential drawback of using enum is that, if later we want to support framework `X` with a new level of `Y`, we need to have another round of API version release to add `Y` into the enum list.
Agree with portal part experience. It should be able to decouple with enum or not, that string can achieve dropdown box as well (it true that not that intuitive as enum)

Reply:
If `string` type can achieve dropdown box, I think using `string` type is more flexible than `enum` type.

Reply:
Another question: Portal could use enum type, but CLI not. should we do the validation (only support X/Y/Z level) on the backend? If not do it, customer could set everything they want, and the log level change won't take effect.

Reply:
We could add the validation at the CLI side as well, to forbid invalid values and let users fail fast.

Reply:
For the portal experience of log level selection, it's fine to make it a dropdown no matter it's `string` or `enum` we will go with, but we should prefer `string` for readability.
For the validation, this should be part of the basic validation in control plane, along with feature flag checks. And yes we should implement such validation in portal and CLI as well for fail fast.

Reply:
thx for your comments, guys. we will use string type.